conda_smithy/tests/test_feedstock_io.py

```diff
@@ -0,0 +1,710 @@
+        ]
+
+        set_mode = functools.reduce(op.or_, perms)
+
+        fio.set_mode_file(filename, set_mode)
+
+        file_mode = os.stat(filename).st_mode
+
+        self.assertEqual(file_mode & set_mode, set_mode)
+
+
+    def test_set_mode_file_2(self):
+        filename = "test.txt"
+
+        with open(filename, "w") as fh:
+            fh.write("")
+
+        self.repo.index.add([filename])
+
+        perms = [
+            stat.S_IWUSR,
+            stat.S_IXUSR,
+            stat.S_IRUSR,
+            stat.S_IXGRP,
+            stat.S_IRGRP,
+            stat.S_IROTH
+        ]
+
+        set_mode = functools.reduce(op.or_, perms)
+
+        fio.set_mode_file(filename, set_mode)
+
+        file_mode = os.stat(filename).st_mode
+
+        self.assertEqual(file_mode & set_mode, set_mode)
+
+
+    def test_write_file_1(self):
+        filename = "test.txt"
+        filename = os.path.join(self.tmp_dir, filename)
+
+        write_text = "text"
+        with fio.write_file(filename) as fh:
+            fh.write(write_text)
+
+        read_text = ""
+        with open(filename, "r") as fh:
+            read_text = fh.read()
+
+        self.assertEqual(write_text, read_text)
+
+        blob = next(self.repo.index.iter_blobs(BlobFilter(filename)))[1]
+        read_text = blob.data_stream[3].read().decode("utf-8")
+
+        self.assertEqual(write_text, read_text)
+
+
+    def test_write_file_2(self):
+        filename = "test.txt"
+
+        write_text = "text"
+        with fio.write_file(filename) as fh:
+            fh.write(write_text)
+
+        read_text = ""
+        with open(filename, "r") as fh:
+            read_text = fh.read()
+
+        self.assertEqual(write_text, read_text)
+
+        blob = next(self.repo.index.iter_blobs(BlobFilter(filename)))[1]
+        read_text = blob.data_stream[3].read().decode("utf-8")
+
+        self.assertEqual(write_text, read_text)
+
+
+    def test_write_file_3(self):
+        filename = "dir1/dir2/test.txt"
+
+        write_text = "text"
+        with fio.write_file(filename) as fh:
+            fh.write(write_text)
+
+        read_text = ""
+        with open(filename, "r") as fh:
+            read_text = fh.read()
+
+        self.assertEqual(write_text, read_text)
+
+        blob = next(self.repo.index.iter_blobs(BlobFilter(filename)))[1]
+        read_text = blob.data_stream[3].read().decode("utf-8")
+
+        self.assertEqual(write_text, read_text)
+
+
+    def test_touch_file_1(self):
+        filename = "test.txt"
+        filename = os.path.join(self.tmp_dir, filename)
+
+        fio.touch_file(filename)
+
+        read_text = ""
+        with open(filename, "r") as fh:
+            read_text = fh.read()
+
+        self.assertEqual("", read_text)
+
+        blob = next(self.repo.index.iter_blobs(BlobFilter(filename)))[1]
+        read_text = blob.data_stream[3].read().decode("utf-8")
+
+        self.assertEqual("", read_text)
+
+
+    def test_touch_file_2(self):
```

Comment:
`test_touch_file_1` and `test_touch_file_2` are basically the same since `os.chdir(self.tmp_dir)` is done during `setUp`, so can we remove one of them?